airbyte-integrations/connectors/source-sendgrid/source_sendgrid/streams.py

```diff
@@ -196,6 +198,34 @@ def initial_path() -> str:
         return "templates"
 
 
+class Messages(SendgridStream, SendgridStreamIncrementalMixin):
+    """
+    https://docs.sendgrid.com/api-reference/e-mail-activity/filter-all-messages
+    """
+
+    data_field = "messages"
+    cursor_field = "last_event_time"
+    primary_key = "msg_id"
+    limit = 1000
+
+    def request_params(self, stream_state: Mapping[str, Any], **kwargs) -> MutableMapping[str, Any]:
+        time_filter_template = "%Y-%m-%dT%H:%M:%SZ"
+        params = super().request_params(stream_state=stream_state, **kwargs)
+        if type(params["start_time"]) == int:
```

Comment:
```suggestion
        if isinstance(params["start_time"], int):
```
This is more robust.